viewer.py

```diff
@@ -177,13 +186,16 @@ def draw_blocks(game_size, grid, cursor, selected, x_offset=0, y_offset=0):
     draw_blocks(dimensions, grid, newgame_json["cursor"], False)
 
     game_speed = newgame_json["game_speed"]
+    pygame.display.set_caption("Rush Hour - GAME STARTED")
 
     while True:
         pygame.display.update()
 
         pygame.event.pump()
-        if pygame.key.get_pressed()[pygame.K_ESCAPE]:
-            asyncio.get_event_loop().stop()
+        for event in pygame.event.get():
```

Comment:
Isto vai consumir todos os eventos e tu só processas 1....

Reply:
Eu preciso do for para verificar se o utilizador pediu para fechar a janela ou se pressionou ESCAPE.

Reply:
No viewer não ha mais eventos, mas o pump() então não está la a fazer nada...

https://www.pygame.org/docs/ref/event.html#pygame.event.get

Reply:
Para cada janela do pygame existe um evento especial chamado QUIT que é evocado aquando do fecho da janela por parte do OS. Por exemplo quando, no windows, se fecha uma janela na cruz do topo direito.

Reply:
Ninguém disse o contrário :) 

O `event.get` chama o `event.pump`, o que eu disse é que se usas o `event.get` a linha anterior do `event.pump` não faz lá nada... pode ser apagada